docker-compose.stack.yml

```diff
@@ -0,0 +1,138 @@
+
+services:
+  pgpool:
+    image: smirart/pgpool:latest
+    ports:
+      - "5432:5432"
+    volumes:
+      - ./pgpool.conf:/etc/pgpool/pgpool.conf
+    restart: always
+    networks:
+      - db-b
+  db-gtm-1:
+    <<: *shared_image
+    environment:
+      - PG_HOST=0.0.0.0
+      - PG_NODE=gtm_1
+      - PG_PORT=6666
+    command: docker-cmd-gtm
+    entrypoint: docker-entrypoint-gtm
+    volumes:
+      - db-gtm-1-data:/var/lib/postgresql
+    networks:
+      - db-a
+    healthcheck:
+      test: ["CMD", "docker-healthcheck-gtm"]
+      interval: 30s
+      timeout: 10s
+      retries: 1
+  db-coord-1:
+    <<: *shared_image
+    environment:
+      - PG_GTM_HOST=db-gtm-1
+      - PG_GTM_PORT=6666
+      - PG_HOST=0.0.0.0
+      - PG_NODE=coord_1
+      - PG_PORT=5432
+      - WAIT_HOSTS=db-gtm-1:6666
+      - WAIT_BEFORE_HOSTS=20
+      - WAIT_HOSTS_TIMEOUT=60
+    command: sh -c "/wait && docker-cmd-coord"
+    entrypoint: docker-entrypoint-coord
+    volumes:
+      - db-coord-1-data:/var/lib/postgresql
+    networks:
+      - db-a
+      - db-b
+    healthcheck:
+      test: ["CMD", "docker-healthcheck-coord"]
+      interval: 30s
+      timeout: 10s
+      retries: 2
+  db-coord-2:
+    <<: *shared_image
+    environment:
+      - PG_GTM_HOST=db-gtm-1
+      - PG_GTM_PORT=6666
+      - PG_HOST=0.0.0.0
+      - PG_NODE=coord_2
+      - PG_PORT=5432
+      - WAIT_HOSTS=db-gtm-1:6666
+      - WAIT_BEFORE_HOSTS=20
+      - WAIT_HOSTS_TIMEOUT=60
+    command: sh -c "/wait && docker-cmd-coord"
+    entrypoint: docker-entrypoint-coord
+    volumes:
+      - db-coord-2-data:/var/lib/postgresql
+    networks:
+      - db-a
+      - db-b
+    healthcheck:
+      test: ["CMD", "docker-healthcheck-coord"]
+      interval: 30s
+      timeout: 10s
+      retries: 2
+  db-data-1:
+    <<: *shared_image
+    environment:
+      - PG_GTM_HOST=db-gtm-1
+      - PG_GTM_PORT=6666
+      - PG_HOST=0.0.0.0
+      - PG_NODE=data_1
+      - PG_PORT=5432
+      - WAIT_HOSTS=db-gtm-1:6666
+      - WAIT_BEFORE_HOSTS=20
+      - WAIT_HOSTS_TIMEOUT=60
+    command: sh -c "/wait && docker-cmd-data"
+    entrypoint: docker-entrypoint-data
+    volumes:
+      - db-data-1-data:/var/lib/postgresql
+    networks:
+      - db-a
+    healthcheck:
+      test: ["CMD", "docker-healthcheck-data"]
+      interval: 30s
+      timeout: 10s
+      retries: 2
+  db-data-2:
+    <<: *shared_image
+    environment:
+      - PG_GTM_HOST=db-gtm-1
+      - PG_GTM_PORT=6666
+      - PG_HOST=0.0.0.0
+      - PG_NODE=data_2
+      - PG_PORT=5432
+      - WAIT_HOSTS=db-gtm-1:6666
+      - WAIT_BEFORE_HOSTS=20
+      - WAIT_HOSTS_TIMEOUT=60
+    command: sh -c "/wait && docker-cmd-data"
+    entrypoint: docker-entrypoint-data
+    volumes:
+      - db-data-2-data:/var/lib/postgresql
+    networks:
+      - db-a
+    healthcheck:
+      test: ["CMD", "docker-healthcheck-data"]
+      interval: 30s
+      timeout: 10s
+      retries: 2
+
+volumes:
+  db-gtm-1-data: {}
```

Comment:
Any reason why all these volumes are named differently to in the other files? I see c59b951a2ad88ecba106f267b78419fc3197ed82 , but I'm not really sure about the rationale… Again, it doesn't really matter what the volumes are called—but I'm curious about the inconsistency.

Reply:
It's just my practice what I've used and seen in another projects. Just for me it's convenient to separate volumes and names of containers by this suffix.

Reply:
Okay. So it's your own convention. I don't mind too much, as this is a separate file—but I note that you're using a convention from elsewhere over the one already here. But it doesn't really matter.